crates/subspace-networking/src/connected_peers.rs

```diff
@@ -405,12 +400,12 @@ impl NetworkBehaviour for Behaviour {
                     let range = 0..(self.config.dialing_peer_batch_size as usize)
                         .min(self.peer_cache.len());
 
-                    let peer_addresses = self.peer_cache.drain(range).collect::<Vec<_>>();
+                    let peer_addresses = self.peer_cache.drain(range);
 
-                    for peer_address in peer_addresses {
-                        self.known_peers.entry(peer_address.0).or_insert_with(|| {
-                            PeerDecision::PendingConnection {
-                                peer_address: peer_address.clone(),
+                    for (peer_id, address) in peer_addresses {
+                        self.known_peers.entry(peer_id).or_insert_with(|| {
+                            PeerConnectionState::Connecting {
+                                peer_address: address,
                             }
                         });
                     }
```

Comment:
Hm, it is actually not guaranteed that the same address is used for multiple connections (I have Multi-WAN at home for example). Maybe we can store a vector of addresses here and that way we implicitly also get the number of connections?

Reply:
Discussed offline, I removed `until` from `Connecting` state.